src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -814,17 +823,17 @@ Real SpatialPooler::avgConnectedSpanForColumnND_(UInt column) const {
 }
 
 
-void SpatialPooler::adaptSynapses_(const UInt inputVector[], //TODO make sparse
-                                   const vector<UInt> &activeColumns) {
+void SpatialPooler::adaptSynapses_(SDR &input,
+                                   SDR &active) {
   vector<Real> permChanges(numInputs_, -1 * synPermInactiveDec_);
+  const auto &inputVector = input.getDense();
   for (UInt i = 0; i < numInputs_; i++) {
     if (inputVector[i] > 0) {
```

Comment:
the goal is to use sparse where possible, I think this is an ideal example. 
Convert input to sparse `input.getSparse()`, for-loop is much shorter, you can skip the `if`